content/influxdb/cloud-serverless/process-data/downsample.md

```diff
@@ -0,0 +1,354 @@
+---
+title: Downsample data stored in InfluxDB
+description: >
+  ...
+menu:
+  influxdb_cloud_serverless:
+    name: Downsample data
+    parent: Process data
+weight: 101
+related:
+  - /influxdb/cloud-serverless/query-data/sql/aggregate-select/, Aggregate or apply selector functions to data (SQL)
+---
+
+Query and downsample time series data stored in InfluxDB and write the
+downsampled data back to InfluxDB.
+
+This guide uses [Python](https://www.python.org/) and the
+[InfluxDB v3 Python client library](https://github.com/InfluxCommunity/influxdb3-python).
+and assumes you have already
+[setup your Python project and virtual environment](/influxdb/cloud-serverless/query-data/sql/execute-queries/python/#create-a-python-virtual-environment).
+
+- [Install dependencies](#install-dependencies)
+- [Prepare InfluxDB buckets](#prepare-influxdb-buckets)
+- [Create InfluxDB clients](#create-influxdb-clients)
+- [Query InfluxDB](#query-influxdb)
+  - [Define a query that performs time-based aggregations](#define-a-query-that-performs-time-based-aggregations)
+  - [Execute the query](#execute-the-query)
+- [Write the downsampled data back to InfluxDB](#write-the-downsampled-data-back-to-influxdb)
+- [Full downsampling script](#full-downsampling-script)
+
+## Install dependencies
+
+Use `pip` to install the following dependencies:
+
+- `influxdb_client_3`
+- `pandas`
+
+```sh
+pip install influxdb3-python pandas
+```
+
+## Prepare InfluxDB buckets
+
+The downsampling process involves two InfluxDB buckets:
```

Comment:
I wonder if we explain here why we use two buckets rather than 2 measurements. So they understand the benefit of retention policies.

Reply:
Good call. I'll add it.